src/app/stores/migrator_baseline.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { AppStoreState } from "./app_store";
+import {
+  ChartElementState,
+  ParentMapping,
+  PlotSegment,
+  PlotSegmentState,
+  ValueMapping
+} from "../../core/specification";
+import {
+  GuideAxis,
+  GuideClass,
+  GuideAttributeNames
+} from "../../core/prototypes/guides";
+import { Specification, uniqueID } from "../../core";
+import { isType } from "../../core/prototypes";
+import {
+  NestedChartElementClass,
+  NestedChartElementProperties
+} from "../../core/prototypes/marks/nested_chart";
+
+type Chart = Specification.Chart<Specification.ObjectProperties>;
+type ChartState = Specification.ChartState<Specification.AttributeMap>;
+type Element = Specification.ChartElement<Specification.ObjectProperties>;
+type ElementState = Specification.ChartElementState<Specification.AttributeMap>;
+type Mark = Specification.Element<Specification.ObjectProperties>;
+
+interface ElementRef {
+  element: Element;
+  index: number;
+  state: ElementState;
+}
+
+/** Upgrade old versions of chart spec and state to newer version */
+
+export function upgradeGuidesToBaseline(appStoreState: AppStoreState) {
+  upgradeScope(appStoreState.chart, appStoreState.chartState);
+  // TODO are nested charts scopes ?
+
+  return appStoreState;
+}
+
+function upgradeScope(parentElement: Chart, parentState: ChartState) {
+  upgradeChartGuides(parentElement, parentState);
+  upgradeGlyphGuides(parentElement, parentState);
+}
+
+function upgradeChartGuides(parentElement: Chart, parentState: ChartState) {
+  // get chart guides
+  const chartGuideRefs = find(
+    parentElement.elements,
+    parentState.elements,
+    element => element.classID === GuideClass.classID
+  );
+
+  chartGuideRefs.forEach(ref => {
+    const { element, state } = ref;
+
+    // convert mappings to actual values
+    const valueMapping = element.mappings.value as ParentMapping;
+    if (valueMapping && valueMapping.type === "parent") {
```

Comment:
Nitpick: Does it make sense to make this a static variable? `"parent"`

Reply:
Since it's a ParentMapping object, type is strongly types as "parent"